src/elementary_operators.jl

```diff
@@ -191,6 +192,144 @@ function identity_operator(s::IndsNetworkMap; kwargs...)
+    # determine intersection of any points, 
+    # and remove them with the opposite sign
+    for i in 1:length(points)
+      p1, d1 = points[i], dimsl[i]
+      for j in (i + 1):length(points)
+        p2, d2 = points[j], dimsl[j]
+
+        # same dimensions, and no point overlap,
+        # can safely ignore
+        (all(d1 .≈ d2) && !all(p1 .≈ p2)) && continue
+
+        # check if dims are the same. 
+        # If they are, check the corresponding dim
+        ps_ = [p1; p2]
+        ds_ = [d1; d2]
+        order = sortperm(ds_)
+        ps, ds = [ps_[order[1]]], [ds_[order[1]]]
+        for k in 2:length(ds_)
+          if (ds_[order[k]] != ds_[order[k - 1]])
+            push!(ps, ps_[order[k]])
+            push!(ds, ds_[order[k]])
+            continue
+          end
+          # found two matching elements
+          if ps_[k] ≈ ps_[k - 1]
+            continue # added previously
+          else # there's no overap here, continue
+            ps, ds = [], []
+            break
+          end
+        end
+        #(length(Set(ds)) != length(ds)) && continue
+        (length(ds) == 0) && continue
+        push!(overlap_points, Vector(ps))
+        push!(overlap_dims, Vector(ds))
+      end
+    end
+    if length(overlap_points) != 0
+      ψ = ψ + -coeff * delta_xyz(s, overlap_points, overlap_dims; truncate_kwargs...)
+    end
+  end
+
+  return promote_operator ? operator_proj(ψ) : ttn(itensornetwork(ψ))
+end
+
+" Create an operator which maps a function to 0 at all points in xs"
+function map_to_zero_operator(
+  s::IndsNetworkMap, xs::Vector, dims::Vector=[1 for _ in xs]; truncate_kwargs...
+)
+  return delta_kernel(
+    s,
+    [[x] for x in xs],
+    [[dim] for dim in dims];
+    remove_overlap=true,
+    coeff=-1,
+    include_identity=true,
+    promote_operator=true,
+    truncate_kwargs...,
+  )
+end
+
+function map_to_zero_operator(s::IndsNetworkMap, x::Number, dim::Int=1; truncate_kwargs...)
+  return map_to_zero_operator(s, [x], [dim]; truncate_kwargs...)
+end
+
+" Map the points xs in dimension dims of the function f to 0"
+function map_to_zeros(
+  f::ITensorNetworkFunction,
+  xs::Vector,
+  dims::Vector=[1 for _ in xs];
+  truncate_kwargs=(;), # for map_operator
+  kwargs..., # for operate
+)
+  s = indsnetworkmap(f)
+  zero_op = map_to_zero_operator(s, xs, dims; truncate_kwargs...)
+  return operate(zero_op, f; kwargs...)
+end
+
+function map_to_zeros(f::ITensorNetworkFunction, x::Number, dim::Int=1; kwargs...)
+  return map_to_zeros(f, [x], [dim]; kwargs...)
+end
+
+""" Create an operator which projects into a constant plane """
+function const_plane_op(s::IndsNetworkMap, xs::Vector, dims::Vector; truncate_kwargs...)
+  return delta_kernel(
+    s,
+    [[x] for x in xs],
+    [[dim] for dim in dims];
+    remove_overlap=true,
+    coeff=1,
+    include_identity=false,
+    promote_operator=true,
+  )
+end
+
+function const_plane_op(s::IndsNetworkMap, xs::Vector, dim::Int; truncate_kwargs...)
```

Comment:
Seems to me `const_plane_op` is not used in any of the tests and is also just a re-name of `delta_kernel`. I think we should just remove the definition (also the name is fairly ambiguous imo)